### ISSUE TITLE
bug(semantic): `TSImportEqualsDeclaration` does not create a symbol

### ISSUE THREAD

**Issue by DonIsaac** [oxc-project/oxc#4091]
```ts
import foo = require('./foo') // `foo` is not added to the symbol table
const bar = require('./bar')  // but `bar` is
```
[Playground Link](https://oxc-project.github.io/oxc/playground/?code=3YCAAIC6gICAgICAgIC0G8rnONK881UOMhQe0DpPTER984qKiS5%2F6hUUcj4033g4LUHKh5RPzqSm3ZQEfyZEIIA%3D)

This causes a false positive in `jsx-no-undef` ([example](https://oxc-project.github.io/oxc/playground/?code=3YCAAIC%2BgICAgICAgIC0G8rnONK88wEEqZ%2F5VIczOSgc6%2BPeK1FjseOeECpGxVxbCXZTvGMZ9beVDbp35khuOGcHwr6%2BebswOrdVWn89eMCA))

```tsx
import Foo = require('./bar')

const Component = () => <Foo />
```
